### PR TITLE
Make Game#check_answer private to prevent cheating

### PR DESCRIPTION
**Pull request by chrisccerami**
Pull Request in response to #20. Made check_answer method of Game class private to prevent the user from overwriting it while playing.

### CODE REVIEW

**Comment by vikram7** [comment]
I'm still able to recreate the bug locally.

screen shot 2015-06-01 at 11 07 34 pm

screen shot 2015-06-01 at 11 07 44 pm

Running `def self.check_answer(current,input,target);true;end` failed the first time, but was treated as 'correct' each time thereafter.

**Comment by chrisccerami** [comment]
@vikram7 This is pretty out of date. Any interest still in merging or should I close it?

**Comment by vikram7** [comment]
@chrisccerami Thanks for the reminder and this PR. If you can take care of the couple syntax related Hound comments, I can merge this in. Also, it looks like the user crashes the program if they try to put in `def self.check_answer(current,input,target);true;end` rather than staying within the program:

screen shot 2015-10-12 at 10 19 13 am

I think it would be nice for the user to be given a splash screen with a finger wagging at them in ASCII (while remaining in the program) instead, but I can open that up as a separate issue.

**Comment by chrisccerami** [comment]
Haha, I can make it so it doesn't crash. I'll let you handle the wagging finger.

**Comment by vikram7** [comment]
Sounds like a plan.

File: lib/rubies/game.rb [hunk 1]
```diff
@@ -198,6 +187,21 @@ def game
...
+        output = routine.call
+        puts "=> #{output}"
+        puts
+        output == target
+      rescue Exception => e
+        eprinter(e)
+        false
+      end
+    end
+
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: This isn't the end of the class body.

File: lib/rubies/game.rb [hunk 2]
```diff
@@ -198,6 +187,25 @@ def game
...
     end
-  end
 
+    private
+
+    def check_answer(current, input, target)
+      begin
+        routine = lambda { eval(input) }
+        output = routine.call
+        puts "=> #{output}"
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: This isn't Rails.

File: lib/rubies/game.rb [hunk 3]
```diff
@@ -198,6 +187,25 @@ def game
       end
       byebye
     end
-  end
 
+    private
+
+    def check_answer(current, input, target)
+      begin
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: That begin is necessary.

File: lib/rubies/game.rb [hunk 4]
```diff
@@ -198,6 +187,25 @@ def game
...
+        routine = lambda { eval(input) }
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: Not here it isn't.

File: lib/rubies/game.rb [hunk 5]
```diff
@@ -198,6 +187,25 @@ def game
...
+        output = routine.call
+        puts "=> #{output}"
+        puts
+        output == target
+      rescue Exception => e
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: Nope.

**Comment by chrisccerami** [comment]
@vikram7 Should be good to go.

**Comment by vikram7** [comment]
Calling "current" now results in an error which doesn't allow a user to get answers right:

```
Sorry, that code resulted in an error:
undefined local variable or method `current' for #<Rubies::Game:0x007fc7ba471f10>
Sorry, that code is incorrect.
```

screen shot 2015-10-12 at 6 35 29 pm

screen shot 2015-10-12 at 6 35 45 pm

Also, at the top of the screen, program internals seem to be presenting themselves:

```
We have some questions for you about this rubies::myhash:
```

versus

```
We have some questions for you about this hash:
```

Which brings to mind that this entire gem needs a test suite.

**Comment by chrisccerami** [comment]
Interesting. I guess I don't understand what current is doing in check_answer.

File: lib/rubies/game.rb [hunk 6]
```diff
@@ -198,6 +187,25 @@ def game
       end
       byebye
     end
-  end
 
+    private
+
+    def check_answer(current, input, target)
```
Comment (chrisccerami) [inline lib/rubies/game.rb]: `current` is apparently necessary, although I still don't see where.

**Comment by chrisccerami** [comment]
Ok, NOW it should be all good.

File: lib/rubies/game.rb (hunk 6 above)
Comment (vikram7) [inline lib/rubies/game.rb]: It's a bit unclear, but `input` contains the string `current` (which the user has to do in order to find the target value).  `eval(input)` it contains the `current` associated with the `current` in the `def play_round` method. `lambda { eval(input) }`  "stores" `current` to be evaluated later. Look at lines 157 and lines 166 in `game.rb` as that might help.

**Comment by vikram7** [comment]
thanks @chrisccerami

**Comment by chrisccerami** [comment]
:+1:

**Pull request merged**